core/src/main/scala/akka/persistence/r2dbc/internal/h2/H2QueryDao.scala

```diff
@@ -24,13 +24,17 @@ private[r2dbc] class H2QueryDao(executorProvider: R2dbcExecutorProvider) extends
   override protected lazy val log: Logger = LoggerFactory.getLogger(classOf[H2QueryDao])
 
   override protected def eventsBySlicesRangeSql(
+      fromSeqNrParam: Boolean,
       toDbTimestampParam: Boolean,
       behindCurrentTime: FiniteDuration,
       backtracking: Boolean,
       minSlice: Int,
       maxSlice: Int): String = {
     // not caching, too many combinations
 
+    def fromSeqNrParamCondition =
+      if (fromSeqNrParam) "AND (db_timestamp != ? OR seq_nr >= ?)" else ""
```

Comment:
Ok. Will add a comment about why the seq number check is only for the same timestamp. And I guess the conditional is not as obvious in this logical form, compared to say `if` in code. This can be read along the lines of `if (db_timestamp == from_timestamp) seq_nr >= from_seq_nr`, with `P -> Q === ~P v Q`

Reply:
Added an explanatory comment. Could also be expressed with `NOT (db_timestamp = ? AND seq_nr < ?)` — not sure if that's more obvious though. Comment notes that seq numbers are per persistence id.